assemblies/assembly-planner-configuration.adoc

```diff
@@ -542,6 +542,145 @@ most appenders, including the console, cause congestion with `debug` and `trace`
 Switch to an async appender to avoid this problem or turn off `debug` logging.
 ====
 
+[[monitoring]]
+[id='monitoring-the-solver-proc_{context}']
+=== Monitoring the solver
+OptaPlanner exposes metrics through https://micrometer.io/[Micrometer], a metrics instrumentation library for Java applications. You can use Micrometer with popular monitoring systems to monitor the OptaPlanner solver.
+
+==== Configuring a Quarkus OptaPlanner application for Micrometer
+To configure your OptaPlanner Quarkus application to use Micrometer and a specified monitoring system, add the Micrometer dependency to the `pom.xml` file.
+
+.Prerequisites
+* You have a Quarkus OptaPlanner application.
+
+.Procedure
+. Add the following dependency to your application's `pom.xml` file where `<MONITORING_SYSTEM>` is a monitoring system supported by Micrometer:
++
+[source]
+----
+<dependency>
+ <groupId>io.quarkus</groupId>
+ <artifactId>quarkus-micrometer-registry-<MONITORING_SYSTEM></artifactId>
+</dependency>
+----
+. To run the application in development mode, enter the following command:
++
+[source]
+----
+mvn compile quarkus:dev
+----
+. To view metrics for your application, enter the following URL in a browser:
++
+[source]
+----
+http://localhost:8080/q/metrics
+----
+
+==== Configuring a Spring Boot OptaPlanner application for Micrometer
+To configure your Spring Boot OptaPlanner  application to use Micrometer and a specified monitoring system, add the Micrometer dependency to the `pom.xml` file.
+
+.Prerequisites
+* You have a Spring Boot OptaPlanner  application.
+
+.Procedure
+. Add the following dependency to your application's `pom.xml` file where `<MONITORING_SYSTEM>` is a monitoring system supported by Micrometer:
++
+[source]
+----
+<dependency>
+ <groupId>org.springframework.boot</groupId>
+ <artifactId>spring-boot-starter-actuator</artifactId>
+</dependency>
+<dependency>
+ <groupId>io.micrometer</groupId>
+ <artifactId>micrometer-registry-<MONITORING_SYSTEM></artifactId>
+</dependency>
+----
+. Add the following line to the application's `application.properties` file:
++
+[source]
+----
+management.endpoints.web.exposure.include=metrics,<MONITORING_SYSTEM>
```

Comment:
The <MONITORING_SYSTEM> here and below are specific to Prometheus; see https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.metrics.export for details on how to set up the various registries. The other registries go to `/actuator/metrics`